### PR TITLE
Enable Regex in Pages Config

### PR DESCRIPTION
**Pull request by Sydney-o9**
Hi Team,

This is a **preliminary** PR to get some thoughts to implement regex on pagesConfig to work on https://github.com/IlusionDev/nextjs-sitemap-generator/issues/50

There are a few different strategies and I am not sure which one you would like to take, if you'd like to have a priority order, etc.

Happy to hear feedback, then, write test.

An example of configuration:

```
    pagesConfig: {
      // Daily
      "/posts": {
        priority: "0.5",
        changefreq: "daily"
      },
      "/posts/.*": {
        priority: "0.5",
        changefreq: "daily"
      },
```

### CODE REVIEW

File: core.js [hunk 1]
```diff
@@ -149,7 +149,26 @@ class SiteMapper {
...
+                return {
+                    pagePath,
+                    outputPath,
+                    priority,
+                    changefreq
+                };
+            }
+            // 1. Generic wildcard configs go first
+            Object.entries(this.pagesConfig).forEach(([key, val]) => {
+                if (key.includes("*")) {
```
Comment (Sydney-o9) [inline core.js:162]: Happy to come up with a better strategy to enable regex support on a particular page config?

Comment (IlusionDev) [reply core.js:162]: interesting, I will check it tomorrow

**Comment by IlusionDev** [comment]
Please make the test and this will be ready

**Review by IlusionDev — CHANGES_REQUESTED** [review]
Add tests please :)

**Pull request merged**